VerilogCodingStyle.md

```diff
@@ -2291,6 +2292,109 @@ always_comb begin
 end
 ```
 
+### Functions and Tasks
+
+The following section applies to synthesizable RTL only.
+
+***In synthesizable RTL use of functions is allowed, provided they are declared
+automatic. Tasks should not be used.***
+
+Functions must be declared outside of a class in either a package or inside a
```

Comment:
I am now wondering if you can declare a function outside of a module/package? i.e

```systemverilog

function automatic logic [2:0] foo(logic [2:0] a);
...
endfunction

module bar();
...
endmodule
```

I think you can but if not this paragraph is redundant.

Reply:
I think it is understood that 'class' is a non-synthesizable construct, so we can omit it here. 

Functions declared in an SV file outside of a module or a package (yes, you can) is visible to the whole compilation unit AFTER that file is compiled. This is potentially adds unnecessary debug overhead. I think it should be limited to within the module / pkg / interface so that the scope is clear.

Reply:
Just a quick note on this (will come back with
further review afterwards), it is legal to put a function
"floating" outside a module and class. Probably good
to leave this guidance here.

Reply:
recommend resolving this conversation and leaving as is.